src/github.com/getlantern/enproxy/conn_impl.go

```diff
@@ -38,34 +37,52 @@ func Dial(addr string, config *Config) (*Conn, error) {
 		return nil, fmt.Errorf("Unable to dial proxy: %s", err)
 	}
 
+	go c.processWrites()
+	go c.processReads()
```

Comment:
I now start the goroutines only after dialing successfully.  This is good for if the dial on line 35 fails since we don't have to stop those goroutines.